### PR TITLE
Unhardcode bio_noise, add allow effect_on_conditions to queue other effect_on_conditions

### PR DESCRIPTION
**Pull request by Ramza13**
#### Summary
None

#### Purpose of change
Unhardcode bio_noise, add more power to dialog system/effect_on_conditions

#### Describe the solution
Add several new dialog functions: condition for if the player or npc is deaf,ability to generate a sound, and ability to queue other effect_on_conditions.

### CODE REVIEW

File: doc/NPCs.md [hunk 1]
```diff
@@ -559,7 +560,7 @@ Effect | Description
...
 `sound_effect: sound_effect_id_string`, *optional* `sound_effect_variant: variant_string`, *optional* `outdoor_event: outdoor_event`,*optional* `volume: volume_int`  | Will play a sound effect of id `sound_effect_id_string` and variant `variant_string`. If `volume_int` is defined it will be used otherwise 80 is the default. If `outdoor_event`(defaults to false) is true this will be less likely to play if the player is underground.
-
+`set_queue_effect_on_condition: effect_on_condition_array`, (*optional* `time_in_future_min: time_in_future_min_int`,`time_in_future_max: time_in_future_max_int` | Will queue up all members of the `effect_on_condition_array`.  Members should either be the id of an effect_on_condition or an inline effect_on_condition. Members will be run between `time_in_future_min_int` and `time_in_future_max_int` seconds in the future. If these are zero(their default value) the eocs will happen instantly.
```
Comment (actual-nh) [inline doc/NPCs.md:563]:
```suggestion
`set_queue_effect_on_condition: effect_on_condition_array`, (*optional* `time_in_future_min: time_in_future_min_int`,`time_in_future_max: time_in_future_max_int` | Will queue up all members of the `effect_on_condition_array`.  Members should either be the id of an effect_on_condition or an inline effect_on_condition. Members will be run between `time_in_future_min_int` and `time_in_future_max_int` seconds in the future. If these are zero (their default value), the eocs will happen instantly.
```

File: doc/NPCs.md [hunk 2]
```diff
@@ -489,6 +489,7 @@ Effect | Description
 `u_mod_pain: pain_int`<br/>`npc_mod_pain: pain_int` | Your character or the NPC will have `pain_int` added or subtracted from its pain.
 `u_add_wet: wet_int`<br/>`npc_add_wet: wet_int` | Your character or the NPC will be wet `wet_int` as if they were in the rain.
 `u_add_power: power_energy`<br/>`npc_add_power: power_energy` | Your character or the NPC will have `power_energy` added or subtracted from its bionic power.
+`u_make_sound, npc_make_sound: message_string`, `volume: volume_int`, `type: type_string`,  | A sound of description `message_string` will be made at your character or the NPC's location of volume `volume_int` and type `type_string`. Possible types are: background, weather, music, movement, speech, electronic_speech, activity, destructive_activity, alarm, combat,    alert, or order
```
Comment (actual-nh) [inline doc/NPCs.md:492]:
```suggestion
`u_make_sound, npc_make_sound: message_string`, `volume: volume_int`, `type: type_string`,  | A sound will be made at your character's or the NPC's location of description `message_string`, volume `volume_int`, and type `type_string`. Possible types are: background, weather, music, movement, speech, electronic_speech, activity, destructive_activity, alarm, combat, alert, and order.
```

File: src/effect_on_condition.cpp [hunk 3]
```diff
@@ -59,6 +61,27 @@ void effect_on_condition::load( const JsonObject &jo, const std::string & )
...
+            jv.throw_error( "Inline effect_on_condition " + inline_id +
+                            " cannot be created as an effect_on_condition already has this id." );
+        }
+
+        effect_on_condition inline_eoc;
+        inline_eoc.load( jv.get_object(), src, inline_id );
+        effect_on_condition_factory.insert( inline_eoc );
+        return effect_on_condition_id( inline_id );
+    } else {
+        jv.throw_error( "effect_on_condition needs to be either string or effect_on_condition object." );
```
Comment (actual-nh) [inline src/effect_on_condition.cpp:81]:
```suggestion
        jv.throw_error( "effect_on_condition needs to be either a string or an effect_on_condition object." );
```

File: data/json/effect_on_condition.json [hunk 4]
```diff
@@ -87,5 +87,35 @@
...
+    "deactivate_condition": { "not": { "u_has_bionics": "bio_noise" } },
+    "effect": [
+      { "message": "bio noise test.", "type": "bad" },
+      { "u_make_sound": "Crackle!", "volume": 60, "type": "movement" },
+      {
+        "set_queue_effect_on_condition": [
+          {
+            "condition": { "not": "u_is_deaf" },
+            "effect": [
+              { "message": "A bionic emits a crackle of noise!", "type": "bad" },
```
Comment (actual-nh) [inline data/json/effect_on_condition.json:106]:
```suggestion
              { "message": "Your faulty bionic emits a crackle of noise!", "type": "bad" },
```
To be clear which and whose bionic.

**Review by actual-nh — COMMENTED** [review]
BTW, shouldn't the "bio_drain" eoc be checking for whether the character has 25kJ, not 24kJ?

**Comment by Ramza13** [comment]
Reminder to self: this needs to have the code to generate inline eoc names updated, right now updating a save game version could change their function.

**Comment by Ramza13** [comment]
Fixed save issue by requiring an id,

**Pull request merged**